moc_openshift.py

```diff
@@ -340,6 +359,59 @@ def update_user_role_project(
             mimetype="application/json",
         )
 
+    def replace_moc_quota(self, project_name, new_quota):
+        """This will delete all resourcequota objects in a project and create new ones based on the new_quota specification"""
+        quota_def = self.get_quota_definitions()
+        if "QuotaMultiplier" in new_quota["Quota"]:
+            quota_multiplier = new_quota["Quota"]["QuotaMultiplier"]
+            for quota in quota_def:
+                quota_def[quota]["value"] = (
+                    quota_def[quota]["coefficient"] * quota_multiplier
+                    + quota_def[quota]["base"]
+                )
+                if "units" in quota_def[quota]:
+                    quota_def[quota]["value"] = (
+                        str(quota_def[quota]["value"]) + quota_def[quota]["units"]
+                    )
+        # RBB: flesh out
+        # else:
+        #
+        # RBB  need to overwrite the value in the quotadef with the ones from the new_quota
+
+        delete_resp = self.delete_moc_quota(project_name)
+        if delete_resp.status_code not in [200, 201]:
+            return Response(
+                response="Unable to delete current quotas in {project_name}\n {delete_resp.status}",
+                status=delete_resp.status_code,
+                mimetype="application/json",
+            )
+        create_resp = self.create_shift_quotas(project_name, quota_def)
+        if create_resp.status_code not in [200, 201]:
+            return Response(
+                response="Unable to define Quotas",
+                status=400,
```

Comment:
why don't we return `create_resp.status_code` instead of our custom 400 here?

Reply:
I think it comes done to what OpenShift considers an error and what we consider an error.  For example, deleting a user that doesn't exist return a status_code of 200 (it's correct, the user doesn't exist after the command), however we would like to consider this an error as we should know who has accounts on the system.  If we delete a account twice - or the more likely possibility, if someone deletes the account, it indicates that a problem occurred that we should track down.  I was using regular expressions to detect this error - the command indicates success why verbalizing that there was nothing to delete.

Now for that one in particular, I forget what error it was sometimes passing back, but it was determined that we did care what openshift was dealing with, we only cared that our command didn't run, and so we passed back a 400.

Reply:
>  I forget what error it was sometimes passing back

Moving forward we should be adding small comments to the code in a case like this.

Reply:
I'll put a blanket comment at the class level, as little comments would need to be all class functions.